### PR TITLE
[SMALLFIX] Enforce Mount does not exist in Alluxio

### PR DESCRIPTION
**Pull request by calvinjia**
I've opened https://alluxio.atlassian.net/browse/ALLUXIO-1830 which should clean up this logic when implemented.

### CODE REVIEW

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 1]
```diff
@@ -1544,6 +1544,20 @@ public void mount(AlluxioURI alluxioPath, AlluxioURI ufsPath)
     MasterContext.getMasterSource().incMountOps(1);
     synchronized (mInodeTree) {
       checkPermission(FileSystemAction.WRITE, alluxioPath, true);
+
+      // Check that the Alluxio Path does not exist
+      boolean pathExists = false;
+      try {
+        mInodeTree.getInodeByPath(alluxioPath);
```
Comment (yupeng9) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: Other than using exception handling for the logic control, shall we add a method checking the existence?

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java [hunk 2]
```diff
@@ -1544,6 +1544,20 @@ public void mount(AlluxioURI alluxioPath, AlluxioURI ufsPath)
...
+        pathExists = true;
+      } catch (InvalidPathException e) {
+        // Expected, continue
+      }
+      if (pathExists) {
```
Comment (yupeng9) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: This is an important check, so could you add a test for it?

**Comment by yupeng9** [comment]
Thanks for adding this check. I left some comments.

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 1 above)
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: See my original comment

File: core/server/src/main/java/alluxio/master/file/FileSystemMaster.java (hunk 2 above)
Comment (calvinjia) [inline core/server/src/main/java/alluxio/master/file/FileSystemMaster.java]: I've created ALLUXIO-1831 which will go to master.

**Comment by yupeng9** [comment]
Thanks. LGTM

**Pull request merged**